hyperspy/axes.py

```diff
@@ -826,6 +999,103 @@ def _on_scale_changed(self):
     def _on_offset_changed(self):
         self.events.any_axis_changed.trigger(obj=self)
 
+    def convert_units(self, axes=None, units=None, same_units=True,
+                      factor=0.25):
+        """ Convert the scale and the units of the selected axes. If the unit 
+        of measure is not supported by the pint library, the scale and units 
+        are not changed.
+
+        Parameters
+        ----------
+        axes : {int | string | iterable of `DataAxis` | None}
+            Default = None
+            Convert to a convenient scale and units on the specified axis.
+            If int, the axis can be specified using the index of the
+            axis in `axes_manager`.
+            If string, argument can be `navigation` or `signal` to select the
+            navigation or signal axes. The axis name can also be provided.
+            If `None`, convert all axes.
+        units : {list of string of the same length than axes | str | None}
+            Default = None
+            If list, the selected axes will be converted to the provided units.
+            If str, the navigation or signal axes will be converted to the 
+            provided units.
+            If `None`, the scale and the units are converted to the appropriate 
+            scale and units to avoid displaying scalebar with >3 digits or too 
+            small number. This can be tweaked by the `factor` argument.
+        same_units : bool
+            If `True`, force to keep the same units if the units of
+            the axes differs. It only applies for the same kind of axis, 
+            `navigation` or `signal`. By default the converted units of the 
+            first axis is used for all axes. If `False`, convert all axes 
+            individually.
+        factor : float
```

Comment:
As ``factor`` appears in many docstrings, it could be extracted into a single string and used where needed.